optaplanner-core/src/test/java/org/optaplanner/core/api/score/buildin/bendable/BendableScoreHolderTest.java

```diff
@@ -46,12 +46,19 @@ public void addConstraintMatch(boolean constraintMatchEnabled) {
         RuleContext ruleContext3 = createRuleContext("scoreRule3");
         scoreHolder.addSoftConstraintMatch(ruleContext3, 0, -30); // Rule match added
         scoreHolder.addSoftConstraintMatch(ruleContext3, 0, -3); // Rule match modified
+        scoreHolder.addHardConstraintMatch(ruleContext3, 0, -300); // Rule of different level added
+        scoreHolder.addHardConstraintMatch(ruleContext3, 0, -400); // Rule of different level modified
 
         scoreHolder.addSoftConstraintMatch(createRuleContext("scoreRule4"), 1, -4); // Rule match added
 
-        assertEquals(BendableScore.valueOf(new int[]{-1000}, new int[]{-3, -4}), scoreHolder.extractScore());
+        RuleContext ruleContext5 = createRuleContext("scoreRule5");
+        scoreHolder.addHardConstraintMatch(ruleContext5, 0, -1);
+        scoreHolder.addSoftConstraintMatch(ruleContext5, 0, -1);
```

Comment:
One trick I use is to avoid using the same decimal order twice in a number used in a unit test.
It's far easier to sum 1000, 200, 30 and 4 than the sum of 1, 2, 3 and 4, so therefor it's far easier to verify the validity of a unit test by reading it :)